src/DrDotnet.Profilers/api/clr_profiler_info.rs

```diff
@@ -77,12 +79,75 @@ impl ClrProfilerInfo {
+                    type_props.name
+                }
+            }
+
+            fn handle_generics(&self, type_name_raw: String, metadata: &MetadataImport, td: mdTypeDef, module_id: ModuleID) -> String {
+                if let Some((start, number_str)) = type_name_raw.split_once('`') {
+                    match number_str.parse::<usize>() {
+                        Ok(number) => {
+                            info!("generic type: {}", type_name_raw);
+                            // Todo: Use number to initialize buffer with appropriate size
+                            match metadata.enum_generic_params(td) {
+                                Ok(generic_params) => {
+                                    let generic_type_names = generic_params.iter().map(|generic_param| {
+                                        match metadata.get_generic_params_props(*generic_param) {
+                                            Ok(generic_param_td) => self.get_type_name(module_id, generic_param_td),
+                                            Err(hresult) => {
+                                                warn!("info.get_generic_params_props({}) failed ({:?})", generic_param, hresult);
+                                                "unknown".to_owned()
+                                            }
+                                        }
+                                    }).join(", ");
+                                    format!("{}<{}>", start, generic_type_names)
+                                },
+                                Err(hresult) => {
+                                    warn!("info.enum_generic_params({}) failed ({:?})", td, hresult);
+                                    type_name_raw
+                                }
+                            }
+                        },
+                        Err(_) => {
+                            warn!("Error while parsing number of generic type parameters");
+                            type_name_raw
+                        },
+                    }
+                } else {
+                    type_name_raw
+                }
+            }
+        }
+
         match self.get_module_metadata(module_id, CorOpenFlags::ofRead) {
-            Ok(metadata) =>
-            match metadata.get_type_def_props(td) {
-                Ok(type_props) => type_props.name,
+            Ok(metadata) => match metadata.get_type_def_props(td) {
+                Ok(type_props) => {
+                    // If type is nested in another type, recursively get the name of the parent type to prefix it
+                    let type_name = self.handle_nesting(type_props, &metadata, td, module_id);
+                    // info!("version {}", metadata.get_version_string().unwrap_or_default()); // Just to show that this function works
+                    //let type_name = self.handle_generics(type_name, &metadata, td, module_id); // Somehow this crashes...
```

Comment:
In the meantime I have commented out the code to handle generics to avoid crashes